apps/beeai-cli/src/beeai_cli/api.py

```diff
@@ -50,29 +51,48 @@ def server_process_status(
     return ProcessStatus.not_running
 
 
+async def set_auth_header():
+    token = await load_token()
+    if not token:
+        raise RuntimeError("No token found. Please run `beeai login` first.")
+    if not token.get("id_token"):
+        raise RuntimeError("No id token found. Please run `beeai login` first.")
+    return f"Bearer {token['id_token']}"
```

Comment:
This is not the appropriate token to send, it must be the `access_token`. The server is not the `id_token`'s audience. The server must use the `access_token` to grant access to resources.

If verified identity data is required by the resource server, the token should have `id_token` scope which allows the server to pick up the data on the auth server's `/userinfo` endpoint.

If unverified identity data is sufficient (e.g. user can change their name freely), the client (CLI/UI) should extract the data from the `id_token` and send them as regular request payload. This is fine since `sub/email` for authorization is still taken from the `access_token`.